source/extensions/filters/http/jwt_authn/filter_config.cc

```diff
@@ -29,6 +37,34 @@ void FilterConfigImpl::init() {
           it.first, Verifier::create(it.second, proto_config_.providers(), *this));
     }
   }
+
+  for (const auto& it : proto_config_.requirement_map()) {
+    if (top_requirement_names_for_debug_.size() < TopRequirementNameForDebugSize) {
+      if (top_requirement_names_for_debug_.empty()) {
+        top_requirement_names_for_debug_ = it.first;
+      } else {
+        absl::StrAppend(&top_requirement_names_for_debug_, ",", it.first);
+      }
+    }
+    name_verifiers_.emplace(it.first,
+                            Verifier::create(it.second, proto_config_.providers(), *this));
+  }
+}
+
+std::pair<const Verifier*, std::string>
+FilterConfigImpl::findPerRouteVerifier(const PerRouteFilterConfig& per_route) const {
+  if (per_route.config().bypass()) {
+    return std::make_pair(nullptr, EMPTY_STRING);
+  }
+
+  const auto& it = name_verifiers_.find(per_route.config().requirement_name());
+  if (it != name_verifiers_.end()) {
+    return std::make_pair(it->second.get(), EMPTY_STRING);
+  }
+
+  return std::make_pair(
+      nullptr, absl::StrCat("Wrong requirement_name: ", per_route.config().requirement_name(),
+                            ". Correct names are: ", top_requirement_names_for_debug_));
```

Comment:
How about `The correct name should be one of [r1, r2,...]`

Reply:
Done